plugin/restore/mysql/mysql.go

```diff
@@ -123,6 +132,170 @@ func GetPITRDatabaseName(database string, timestamp int64) string {
+						return fmt.Errorf("cannot remove %s, error: %w", localFileName, err)
+					}
+				} else {
+					// File size match, record it in downloadedIndex
+					downloadedIndex[index] = struct{}{}
+				}
+			}
+		}
+	}
+
+	// 6. download the binlog that not record in downloadedIndex
+	for i, serverFile := range binlogFilesOnServer {
+		if _, ok := downloadedIndex[i]; ok {
+			continue
+		}
+		if err := r.syncBinlog(ctx, instance, saveDir, serverFile); err != nil {
+			return fmt.Errorf("cannot sync binlog %s, error: %w", serverFile.Name, err)
+		}
+	}
+
+	return nil
+}
+
+// SyncLatestBinlog syncs the latest binlog between the instance and `saveDir`
+func (r *Restore) SyncLatestBinlog(ctx context.Context, instance *api.Instance, saveDir string) error {
+	latest, err := r.showLatestBinlogFile(ctx)
+	if err != nil {
+		return err
+	}
+	return r.syncBinlog(ctx, instance, saveDir, *latest)
+}
+
+// syncBinlog syncs the binlog specified by `meta` between the instance and local.
+func (r *Restore) syncBinlog(ctx context.Context, instance *api.Instance, saveDir string, meta mysql.BinlogFile) error {
+	tmpFilePrefix := fmt.Sprintf("tmp_%d_", time.Now().UnixNano())
+	// TODO(zp): support ssl?
+	cmd := exec.CommandContext(ctx, filepath.Join(mysqlbinlogPath, "bin", "mysqlbinlog"),
+		meta.Name,
+		fmt.Sprintf("--read-from-remote-server --host=%s --port=%s --user=%s --password=%s", instance.Host, instance.Port, instance.Username, instance.Password),
+		"--raw",
+		// Note, recheck here when upgrade embedding mysqlbinlog binary
+		// for mysqlbinlog with binlog.000001 --result-file=a_, it will save it as a_binlog.000001
+		fmt.Sprintf("--result-file=%s", filepath.Join(saveDir, tmpFilePrefix)),
+	)
+	cmd.Stdout = os.Stderr
+	cmd.Stderr = os.Stderr
+
+	var runErr error
+	defer func() {
+		tmpFileName := fmt.Sprintf("%s%s", tmpFilePrefix, meta.Name)
+		tmpFilePath := filepath.Join(saveDir, tmpFileName)
+		if runErr != nil {
+			// delete the download binlog if some error occur
+			_ = os.Remove(tmpFilePath)
+			return
+		}
+
+		fi, err := os.Stat(tmpFilePath)
+		if err != nil {
+			// delete the download binlog if cannot stat it
+			_ = os.Remove(tmpFilePath)
+			return
+		}
+		if fi.Size() != meta.Size {
+			// delete the download binlog if filesize not match
+			_ = os.Remove(tmpFilePath)
+			return
+		}
+
+		// no error, rename it
+		_ = os.Rename(tmpFilePath, filepath.Join(saveDir, meta.Name))
+	}()
+
+	if runErr = cmd.Run(); runErr != nil {
+		return fmt.Errorf("cannot run mysqlbinlog, error: %w", runErr)
+	}
+	return nil
+}
+
+// showBinlogFiles returns the metadata of binlogs
+func (r *Restore) showBinlogFiles(ctx context.Context) ([]mysql.BinlogFile, error) {
+	db, err := r.driver.GetDbConnection(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+	// https://dev.mysql.com/doc/refman/8.0/en/show-binary-logs.html
+	stmt := `SHOW BINARY LOGS;`
+	rows, err := db.QueryContext(ctx, stmt)
```

Comment:
```suggestion
	rows, err := db.QueryContext(ctx, "SHOW BINARY LOGS;")
```
We can make it more concise if no string format is needed.